alf/environments/suite_carla.py

```diff
@@ -131,6 +131,118 @@ def _to_numpy_loc(loc: carla.Location):
     return np.array([loc.x, loc.y, loc.z])
 
 
+class WeatherParameters(object):
+    """A class for a set of weather related parameters. Currently it contains
+    all the weather fields from ``carla.WeatherParameters`` except for
+    ``sun_azimuth_angle`` and ``sun_altitude_angle``, which are controlled
+    separately by ``day_length`` in a more realistic way.
+    """
+
+    def __init__(self,
+                 cloudiness=0,
+                 precipitation=0,
+                 precipitation_deposits=0,
+                 wind_intensity=0,
+                 fog_density=0,
+                 fog_distance=0):
+        self.cloudiness = cloudiness  # [0, 100]
+        self.precipitation = precipitation  # [0, 100]
+        self.precipitation_deposits = precipitation_deposits  # [0, 100]
+        self.wind_intensity = wind_intensity  # [0, 100]
+        self.fog_density = fog_density  # [0, 100]
+        self.fog_distance = fog_distance  # [0, 100]
+        self._fields = None
+
+    def get_weather_fields(self):
+        """ Get the list of configurable weather fields
+
+        Returns:
+            A list of strings, each as the name of a configurable field
+        """
+        if self._fields is None:
+            self._fields = [
```

Comment:
since get_weather_fields is always used, it's better to move this to `__init__` to avoid the if statement later.

Reply:
Move it to ``__init__``